cpp/oneapi/dal/test/array.cpp

```diff
@@ -327,6 +327,45 @@ TEST("can wrap const data with queue, offset and deleter") {
     REQUIRE(arr.get_data() == cdata);
     REQUIRE(!arr.has_mutable_data());
 }
+
+TEST("can be created from shared_ptr") {
+    DECLARE_TEST_POLICY(policy);
+    auto& q = policy.get_queue();
+    constexpr std::int64_t count = 3;
+    auto* data = sycl::malloc_shared<float>(count, q);
+    for (std::int64_t i = 0; i < count; ++i) {
+        data[i] = float(i);
+    }
+    std::shared_ptr<float> sdata(data, [&q](auto data) {
+        sycl::free(data, q);
+    });
+
+    auto darr = array<float>{ sdata, count };
```

Comment:
That is not right way to create an array from USM memory. Use ctor with queue parameter instead.